api/filters/paging.go

```diff
@@ -52,22 +52,32 @@ func (*pagingFilter) Name() string {
 func (pf *pagingFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
 	ctx := req.Context()
 	maxItems := req.URL.Query().Get("max_items")
-	limit, err := pf.validateMaxItemsQuery(maxItems)
+	limit, err := pf.parseMaxItemsQuery(maxItems)
 	if err != nil {
 		return nil, err
 	}
-	ctx = context.WithValue(ctx, "limit", limit)
+
+	rawToken := req.URL.Query().Get("token")
+	token, err := pf.parsePageToken(rawToken)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = web.ContextWithPageLimit(ctx, limit)
 	if limit > 0 {
-		ctx, err = query.AddCriteria(ctx, query.LimitResultBy(limit))
+		ctx, err = query.AddCriteria(ctx, query.LimitResultBy(limit+1),
```

Comment:
LimitResultsTo should be a more appropriate name for the method called

Adding +1 seems counterintuitive to what the method name states. Whats the reason?

Reply:
Because if you set the limit as provided in the request there is no way to say afterwards if there is more items or not (if you need to generate and return a token in the response).